src/main/java/Echoa/Parser.java

```diff
@@ -0,0 +1,207 @@
+     */
+    public static LocalDateTime createDateTime(String dateString, String timeString) throws DateTimeParseException {
+        LocalDate date = LocalDate.parse(dateString, DATE_FORMATTER);
+        LocalTime time = LocalTime.parse(timeString, TIME_FORMATTER);
+        return LocalDateTime.of(date, time);
+    }
+
+    /**
+     * The method parses the line for specific instructions.
+     *
+     * @param line line to be parsed.
+     * @return instruction keyword.
+     */
+    public String parseCommand(String line) {
+        String command;
+        line = line.trim();
+        if (line.isBlank()) {
+            command = "";
+        } else if (line.startsWith("list")) {
+            command = "list";
+        } else if (line.startsWith("find")) {
+            command = "find";
+        } else if (line.startsWith("mark")) {
+            command = "mark";
+        } else if (line.startsWith("unmark")) {
+            command = "unmark";
+        } else if (line.startsWith("delete")) {
+            command = "delete";
+        } else if (line.startsWith("todo")) {
+            command = "todo";
+        } else if (line.startsWith("deadline")) {
+            command = "deadline";
+        } else if (line.startsWith("event")) {
+            command = "event";
+        } else if (line.startsWith("bye")) {
+            command = "bye";
+        } else {
+            command = line;
+        }
+        return command;
+    }
+
+    /**
+     * The method parses for indexes in lines with instructions:
+     * mark, unmark and delete.
+     *
+     * @param indexString indexString contains a possible label
+     * @return the index (0-indexed)
+     */
+    public int parseIndex(String indexString) throws InvalidIndexInputException {
+        String[] lineArray = indexString.split("\\s+");
+        int numOfWords = lineArray.length;
+
+        if (numOfWords != 1) {
+            throw new InvalidIndexInputException(indexString);
+        }
+
+        Pattern pattern = Pattern.compile("\\d");
+        Matcher matcher = pattern.matcher(indexString);
+        if (!matcher.find()) {
+            throw  new InvalidIndexInputException(indexString);
+        }
+
+        System.out.println("label: " + Integer.parseInt(indexString));
+        return Integer.parseInt(indexString) - 1;
+    }
+
+    /**
+     * The method searches for tasks in the taskList that contains the specified keyword.
+     *
+     * @param taskList taskList to be searched.
+     * @param keyword keyword to be searched.
+     * @return TaskList containing tasks containing the keyword.
+     */
+    public TaskList parseFindTask(TaskList taskList, String keyword) {
+
+        TaskList tasks = new TaskList();
+
+        for (int i = 0; i < taskList.getSize(); i++) {
+            Task t = taskList.getSpecificTask(i);
+            if (t.getDescription().contains(keyword)) {
+                tasks.addTask(t);
+            }
+        }
+
+        return tasks;
+    }
```

Comment:
I've noticed sometimes your return statement has an extra line break to separate it but other times it does not. try to stay consistent! the idea behind coding standards are so that it looks like all the code was written by one person.